notebooks/applications/spark/Kubectl-access-locally.md

```diff
@@ -0,0 +1,73 @@
+# Kubectl access locally
+
+## Why
+
+Out of the box, you only have `kubectl` via your Terminal from the JupyterLab in the Kubeflow dashboard. Especially when getting logs from pipelines, SparkApplications and more, it makes sense to have `kubectl` configured with correct permission handling. This might be faster and you are able to open more terminals at the same time. It might be also possible to use tools like Lens.
+
+## Requirements
+
+- This manual requires Linux (e.g. Ubuntu) but can be adjusted also for Windows
+- Kubectl is installed
+- You have a user on our Kubeflow platform
+- You have a JupyterLab in the Kubeflow dashboard (in order to extract the token)
+- The JupyterLab has according permissions (is automatically handled via RBAC of the ServiceAccount `kubeflow-editor`)
+
+## Gather information
+
+In a terminal of your JupyterLab, run the following commands.
+
+1. Get the token from your serviceaccount `kubeflow-editor` (**Needs to be done every day!**):
+```
+TOKENNAME=`kubectl get serviceaccount/default-editor -o jsonpath='{.secrets[0].name}'`
+echo $TOKENNAME #Verification
+TOKEN=`kubectl get secret $TOKENNAME -o jsonpath='{.data.token}'| base64 --decode`
+echo $TOKEN #You need it for the setup later
+CA=`kubectl get secret $TOKENNAME -o jsonpath='{.data.ca\.crt}'| base64 --decode`
+echo $CA #You need it for the setup later
+```
+2. Get your namespace and adjust it accordingly in this manual:
+```
+echo $NAMESPACE
+tim-krause
+```
+
+## Setup
+
+1. Insert your token into you local terminal:
+```
+TOKEN=eyJhbGciOiJSUzI1NiIsImtpZCI6IldYcE5NOXhQaXE3X3RlM05SSnI5T3FxME9vMzhDSGhmQ281ZDFfOW9vdkkifQ.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJ0aW0ta3JhdXNlIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImRlZmF1bHQtZWRpdG9yLXRva2VuLXM5MmNsIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImRlZmF1bHQtZWRpdG9yIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiYjY3YTZlOTctNTFjNS00MTE5LTlmMDgtZTZkZjBhMzRiZDFkIiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OnRpbS1rcmF1c2U6ZGVmYXVsdC1lZGl0b3IifQ.Awui61Cw-JlSTONRNEFl-i2sAST0YFIiCbZtPDasRuj9gRZFpNkFMKpgTiRuQPeCeWZ51P-BnWqpQg3W6UW-jjDtmIh2d5yIQn7jK6MesYPdy7hYt-VQKJCh8tFRpSvhYpXLJ_n0B0WfGQYlwqh1ATloNLcSUom2gzjgb1GDXGkKlBvpRsHxCmUFUIWuLJ6XVZrA0XBULa0eaICpzQVXi25CWt53Iv0_379FYXPDYoLpD6lW-ur2ZrzMncM04aQanihFojFOys09dB_i8atX9vtaGkXVasEzLMEGTNKHUbyZyzR5IEEkgWSqctJI-_miyF7k9jgk6OFh9mgnMfhufA
+
+echo $TOKEN #Verification
```

Comment:
Please check if the token here should be the base64 encoded or decoded version. If it must be the encoded string, please remove the decode command in the token extraction example above

Reply:
fixed